src/key/key.js

```diff
@@ -906,3 +675,16 @@ class Key {
 
 export default Key;
 
+/**
+ * Creates a PublicKey or PrivateKey depending on the packetlist in input
+ * @param {PacketList} - packets to parse
+ * @return {Key} parsed key
+ */
+export function createKey(packetlist) {
+  switch (packetlist[0].constructor.tag) {
```

Comment:
I don't feel very strongly about this, but technically speaking it's legal for the first packet to be a Marker packet, for example, and the second packet to be the key packet. Currently we would accept that, I think, this would break that (though I don't think it's very likely to happen in the wild, but it might be more robust to check if there's any private key packet in the list, for example).